### PR TITLE
Switch confirmed_unrooted_slots from Vec<_> to HashSet<_>

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
The container is only used to check for inclusion of slots with
the .contains() method. This method is O(n) on a Vec<_> but O(1) on a
HashSet<_>.

The container should be fairly small given that it is populated with confirmed slots that are newer than the most recent root. So, in normal network conditions, should be around the lockout length. Regardless, still a more proper change to make

### CODE REVIEW

File: ledger/src/blockstore.rs [hunk 1]
```diff
@@ -2348,14 +2348,14 @@ impl Blockstore {
             "blockstore-rpc-api",
             ("method", "get_rooted_transaction_status", String)
         );
-        self.get_transaction_status(signature, &[])
+        self.get_transaction_status(signature, &HashSet::default())
     }
 
     /// Returns a transaction status
     pub fn get_transaction_status(
```
Comment (CriesofCarrots) [inline ledger/src/blockstore.rs:2355]: I was concerned about this change to the signature of a public function, but it seems like nothing is calling this, in the monorepo at least.

**Review by CriesofCarrots — COMMENTED** [review]
Seems like a fair trade-off. Bump me when this is past CI and I'll approve

File: ledger/src/blockstore.rs (hunk 1 above)
Comment (steviez) [reply ledger/src/blockstore.rs:2355]: Yeah, I had a similar thought but monorepo (namely `solana_rpc`) built so went forward with it.

**Comment by steviez** [comment]
> Seems like a fair trade-off. Bump me when this is past CI and I'll approve

Shoot, my bad; I thought I had seen greens before adding you but that obviously wasn't the case. Rebase on tip of master did the trick

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**